### ISSUE TITLE
MissingGasError: Invalid value given "gas: 0x5208, gasPrice: undefined, maxPriorityFeePerGas: 0x2faf080, maxFeePerGas: undefined". Error: "gas" is missing.

### ISSUE THREAD

**Issue by Nakinthorn** [ethereum/web3.js#6276]
const transactionObject = {
    from: accountAddress,
    to: contractAddress,
    data: contract.methods.transfer(recipientAddress, amount).encodeABI(),
    gas: 21000,
    maxPriorityFeePerGas:50000000,
    //maxFeePerGas:500000000

  };
  
  i try to do everything and change all possible key and value
  
  i try to enter gasPrice,maxPriorityFeePerGas,maxFeePerGas. like err say but still err
  
 it always err
 
 MissingGasError: Invalid value given "gas: 0x5208, gasPrice: undefined, maxPriorityFeePerGas: 0x2faf080, maxFeePerGas: undefined". Error: "gas" is missing.

**Comment by luu-alex** [comment]
Hi there, sorry for the confusing error. this error you are getting is misleading and has been reported and will be fixed soon.
For your settings you can just define gas and it should work
```
const transactionObject = {
from: accountAddress,
to: contractAddress,
data: contract.methods.transfer(recipientAddress, amount).encodeABI(),
gas: 21000
};
```

**Comment by Nakinthorn** [comment]
> Hi there, sorry for the confusing error. this error you are getting is misleading and has been reported and will be fixed soon. For your settings you can just define gas and it should work
> 
> ```
> const transactionObject = {
> from: accountAddress,
> to: contractAddress,
> data: contract.methods.transfer(recipientAddress, amount).encodeABI(),
> gas: 21000
> };
> ```

thank you for ur reply
btw , after I set obj value from your code it still err

transactionObject  {
  from: accountAddress,
  to: contractAddress,
 data: contract.methods.transfer(recipientAddress, amount).encodeABI(),
  gas: 21000
}
Signing Error: MissingGasError: Invalid value given "gas: 0x5208, gasPrice: undefined, maxPriorityFeePerGas: undefined, maxFeePerGas: undefined". Error: "gas" is missing.

**Comment by luu-alex** [comment]
What method are you using?

**Comment by luu-alex** [comment]
To fix this issue you need to as well specify gasPrice
```
const transactionObject = {
from: accountAddress,
to: contractAddress,
data: contract.methods.transfer(recipientAddress, amount).encodeABI(),
gas: 21000,
gasPrice: await web3.eth.getGasPrice()
};
```

**Comment by luu-alex** [comment]
Fixed in #6265

**Comment by mkvy** [comment]
> To fix this issue you need to as well specify gasPrice
> 
> ```
> const transactionObject = {
> from: accountAddress,
> to: contractAddress,
> data: contract.methods.transfer(recipientAddress, amount).encodeABI(),
> gas: 21000,
> gasPrice: await web3.eth.getGasPrice()
> };
> ```

thank you, but it gives still:
    gasPrice: await web3.eth.getGasPrice(),
                    ^^^^

SyntaxError: Unexpected identifier
